openapi.yaml

```
@@ -0,0 +1,1791 @@
+              schema:
+                $ref: '#/components/schemas/Error'
+  /pallets/staking/progress:
+    get:
+      tags:
+      - pallets
+      summary: Get a progress report on the chain's staking system.
+      description: Returns information on the progress of key components of the 
+        staking system and estimates of future points of interest. Replaces 
+        `/staking-info` from versions < v1.0.0.
+      operationId: getStakingProgress
+      parameters:
+      - name: at
+        in: query
+        description: Block at which to retrieve a staking progress report at.
+        required: false
+        schema:
+          type: string
+          description: Block identifier, as the block height or block hash.
+          format: unsignedInteger or $hex
+      responses:
+        "200":
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/StakingProgress'
+        "400":
+          description: invalid blockId supplied for at query param
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+  /pallets:
+    get:
+      tags:
+      - pallets
+      summary: Get a list of pallets used in the runtime.
+      description: Returns an array of metadata for pallets that are used in the
+        runtime. Can use `only-names` query param to just return an array of 
+        pallet names.
+      parameters:
+      - name: only-names
+        in: query
+        description: Only return the names of pallets.
+        required: false
+        schema:
+          type: boolean
+      - name: at
+        in: query
+        description: Block at which to retrieve a list of the runtime's pallets 
+          at.
+        required: false
+        schema:
+          type: string
+          description: Block identifier, as the block height or block hash.
+          format: unsignedInteger or $hex
+      responses:
+        "200":
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/PalletsResponses'
+        "400":
+          description: invalid blockId supplied for at query param
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+  /pallets/{palletId}/constants:
+    get:
+      tags:
+      - pallets
+      summary: Get a list of constants for a pallet.
+      description: Returns a list of metadata for constants of the pallet specified
+        by palletId.
+      parameters:
+      - name: palletId
+        in: path
+        description: Id of the pallet to query the constant of.
+        required: true
+        schema:
+          type: string
+      - name: at
+        in: query
+        description: Block at which to retrieve the list of exposed pallet
+          constants at.
```

You get the pattern... will stop commenting
```suggestion
          constants.
```